src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -195,7 +175,7 @@ sealed internal class TdsParserStateObject
         internal SqlErrorCollection _preAttentionWarnings;
 
         volatile private TaskCompletionSource<object> _writeCompletionSource = null;
-        volatile private int _asyncWriteCount = 0;
+        volatile protected int _asyncWriteCount = 0;
```

Comment:
Nit: style-wise, the visibility should come first, e.g. `protected volatile` instead of `volatile protected`